src/game/common/thing/thingtemplate.cpp

```diff
@@ -1143,3 +1144,76 @@ void ThingTemplate::Validate_Audio()
         }
     }
 }
+
+int ThingTemplate::Calc_Cost_To_Build(const Player *player) const
+{
+    if (player != nullptr) {
+        float cost = player->Get_Production_Cost_Change_Percent(Get_Name()) + 1.0f;
+        cost *= player->Get_Production_Cost_Change_Based_On_Kind_Of(m_kindOf);
+        cost *= Get_Build_Cost();
+        return player->Get_Handicap()->Get_Handicap(Handicap::BUILDCOST, this) * cost;
+    } else {
+        return 0;
+    }
+}
+
+int ThingTemplate::Calc_Time_To_Build(const Player *player) const
+{
+    float frame_build_time = (int)(Get_Build_Time() * 30.0f);
```

Comment:
could you replace your C style casts with static casts.

Reply:
fixed.